docs/discussion/lwip_ipv6.md

```diff
@@ -0,0 +1,139 @@
+# LwIP changes for CHIP
+
+LwIP is the network layer used in the CHIP platform. Although it has some good
+IPv6 support, there are areas that are lacking that we should implement for
+CHIP. The recommendations here are listed roughly from most to least important.
+
+## Route Information Options (RIO)
+
+The addressing proposal requires devices to store route options from Route
+Information Options (RIO) sent in router advertisements. This functionality is
+not currently present in upstream LwIP. The patch to add this is relatively
+small, but we may need to upstream this in order to require its use in CHIP. It
+can be added to the CHIP version, but not all platforms use the same middleware.
+
+### Recommendation:
+
+-   write a RIO patch, add to CHIP middleware, upstream to lwip
+    -   Ensure patch is RFC compliant (especially re: expiry)
+
+## DNS
+
+LwIP’s DNS handling isn’t great and breaks down when the router supports
```

Comment:
CHIP doesn't require DNS support, except for time sync NTP, and optional HTTPS support for OTA download, both of which are optional in spec and likely not supported in the very short term roadmap. I would consider this to be a low priority item and we should move it to the bottom of the file/list to make it more obvious it's not a top item.